core/services/aws/commons/ddb/DynamoDbTableOperations.ts

```diff
@@ -34,6 +35,116 @@ export default class DynamoDbTableOperations<
+    try {
+      const { keyConditionExpression, expressionAttributeValues, expressionAttributeNames } =
+        this.buildKeyConditionExpression(queryInput.partitionKeyCondition, queryInput.sortKeyCondition)
+
+      const queryCommandInput: QueryCommandInput = {
+        TableName: this.getTableName(),
+        KeyConditionExpression: keyConditionExpression,
+        ExpressionAttributeValues: expressionAttributeValues,
+        ExpressionAttributeNames: expressionAttributeNames
+      }
+
+      this.applyQueryOptions(queryCommandInput, queryInput.options)
+
+      const result = await this.client.send(new QueryCommand(queryCommandInput))
+
+      return {
+        items: (result.Items ?? []).map(item => this.modelAdapter.toDto(item as DbFields)),
+        lastEvaluatedKey: result.LastEvaluatedKey
+      }
+    } catch (error) {
+      throw new DatabaseError(
+        `Failed to query items from DynamoDB: ${error instanceof Error ? error.message : 'Unknown error'}`,
+        GENERIC_CODES.ERROR
+      )
+    }
+  }
+
+  private applyQueryOptions(queryCommandInput: QueryCommandInput, options?: QueryInput<DbFields>['options']): void {
+    if (options == null) return
+
+    const { indexName, limit, scanIndexForward, exclusiveStartKey, filterExpression, filterExpressionValues } = options
+
+    if ((indexName?.trim()) != null) {
+      queryCommandInput.IndexName = indexName
+    }
+    if ((limit != null) && limit > 0) {
+      queryCommandInput.Limit = limit
+    }
+    if (scanIndexForward !== undefined) {
+      queryCommandInput.ScanIndexForward = scanIndexForward
+    }
+    if ((exclusiveStartKey != null) && Object.keys(exclusiveStartKey).length > 0) {
+      queryCommandInput.ExclusiveStartKey = exclusiveStartKey
+    }
+    if ((filterExpression?.trim()) != null) {
+      queryCommandInput.FilterExpression = filterExpression
+      if ((filterExpressionValues != null) && Object.keys(filterExpressionValues).length > 0) {
+        queryCommandInput.ExpressionAttributeValues = {
+          ...queryCommandInput.ExpressionAttributeValues,
+          ...filterExpressionValues
+        }
+      }
+    }
+  }
+
+  private buildKeyConditionExpression(
+    partitionKeyCondition: QueryCondition<DbFields>,
+    sortKeyCondition?: QueryCondition<DbFields>
+  ): {
+      keyConditionExpression: string;
+      expressionAttributeValues: Record<string, unknown>;
+      expressionAttributeNames: Record<string, string>;
+    } {
+    const expressionAttributeValues: Record<string, unknown> = {}
+    const expressionAttributeNames: Record<string, string> = {}
+
+    const pkName = `#${String(partitionKeyCondition.attributeName)}`
+    const pkValue = `:${String(partitionKeyCondition.attributeName)}`
+    expressionAttributeNames[pkName] = String(partitionKeyCondition.attributeName)
+    expressionAttributeValues[pkValue] = partitionKeyCondition.attributeValue
+
+    let keyConditionExpression = `${pkName} ${partitionKeyCondition.operator} ${pkValue}`
```

Comment:
instead of this let
you could use
```tsx
const baseKeyConditionExpression = `${pkName} ${partitionKeyCondition.operator} ${pkValue}`

    if (sortKeyCondition != null) {
      const skName = `#${String(sortKeyCondition.attributeName)}`
      const skValue = `:${String(sortKeyCondition.attributeName)}`
      expressionAttributeNames[skName] = String(sortKeyCondition.attributeName)
      expressionAttributeValues[skValue] = sortKeyCondition.attributeValue

      switch (sortKeyCondition.operator) {
        case QueryConditionOperator.BEGINS_WITH:
            return {
                keyConditionExpression: `${baseKeyConditionExpression} AND begins_with(${skName}, ${skValue})`,
               expressionAttributeValues,
               expressionAttributeNames
               }
          
         
     ...
      }```

Reply:
updated according to this suggestion, updated at 78987ec55bbffac5d487a6040d415ded5af9e373